Common/Helpers/AzureTableStorageManager.cs

```diff
@@ -10,37 +11,39 @@
 
 namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers
 {
-    public class AzureTableStorageHelper : IAzureTableStorageHelper
+    public class AzureTableStorageManager : IAzureTableStorageManager
     {
-        private readonly string _storageConnectionString;
-        private readonly string _tableName;
-        private CloudTable _table;
+        private readonly ICloudTableProvider _cloudTableProvider;
 
-        public AzureTableStorageHelper(string storageConnectionString, string tableName)
+        public AzureTableStorageManager(string storageConnectionString, string tableName)
         {
-            _storageConnectionString = storageConnectionString;
-            _tableName = tableName;
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(_storageConnectionString);
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storageConnectionString);
             CloudTableClient tableClient = storageAccount.CreateCloudTableClient();
-            _table = tableClient.GetTableReference(_tableName);
+            _cloudTableProvider = new CloudTableProvider(tableClient, tableName);
```

Comment:
Instead of doing a new CloudTableProvider(...) have it injected in as part of the constructor. This way you don't have a direct reference to the CloudTableProvider concrete implementation.